src/inmanta/env.py

```diff
@@ -69,17 +124,106 @@ def install_from_source(cls, paths: List[LocalPackagePath]) -> None:
             raise Exception("install_from_source requires at least one package to install")
         # make sure we only try to install from a local source
         explicit_paths: Iterator[LocalPackagePath] = (
-            LocalPackagePath(path=os.path.join(".", path.path), editable=path.editable) for path in paths
+            LocalPackagePath(path=os.path.join(".", path.path, ""), editable=path.editable) for path in paths
         )
         subprocess.check_call(
             [
-                cls.env_path,
+                cls.python_path,
                 "-m",
                 "pip",
                 "install",
                 *chain.from_iterable(["-e", path.path] if path.editable else [path.path] for path in explicit_paths),
             ]
         )
+        # TODO: other install methods patch pkg_resources.working_set, do we need to do so here as well?
+
+    @classmethod
+    def check(cls, in_scope: Pattern[str], constraints: Optional[List[Requirement]] = None) -> bool:
+        """
+        Check this Python environment for compatible dependencies in installed packages.
+
+        :param in_scope: A full pattern representing the package names that are considered in scope for the installed packages'
+            compatibility check. If constraints are supplied, those are always checked, regardless of this pattern.
+        :param constraints: In addition to checking for compatibility within the environment, also verify that the environment's
+            packages meet the given constraints. This is insterpreted as a list of constraints, not a list of requirements: if
+            one of the constraints' package is not installed the check succeeds.
+        """
+
+        def check_installed() -> List[str]:
+            process: subprocess.CompletedProcess = subprocess.run(
```

Comment:
This is called on every compile, it would be good to investigate if we can do this faster via, e.g. pkg_resources


(logically, all this info has to be in the loader already, so we shouldn't have to fork for it)

Reply:
Done

Reply:
This one did really improve test time (not just because of the Jenkins runner) to the point where this branch's tests now run only a bit slower than those on master, which could be explained by there being more of them.